cancel-in-progress/parent_workflow.go

```diff
@@ -0,0 +1,109 @@
+package cancel_in_progress
+
+import (
+	"errors"
+	"go.temporal.io/sdk/log"
+	"go.temporal.io/sdk/workflow"
+)
+
+const ParentWorkflowSignalName = "parent-workflow-signal"
+
+func SampleParentWorkflow(ctx workflow.Context) (result string, err error) {
+	logger := workflow.GetLogger(ctx)
+
+	var message string
+
+	reBuildSignalChan := workflow.GetSignalChannel(ctx, ParentWorkflowSignalName)
+
+	// This will not block because the workflow is started with a signal transactional.
+	reBuildSignalChan.Receive(ctx, &message)
+
+	cwo := workflow.ChildWorkflowOptions{
+		WaitForCancellation: true,
+	}
+	ctx = workflow.WithChildOptions(ctx, cwo)
+
+	var isProcessingDone = false
+
+	for !isProcessingDone {
+		ctx, cancelHandler := workflow.WithCancel(ctx)
+
+		// it is important to re-execute the child workflow after every signal
+		// because we might have cancelled the previous execution
+		childWorkflowFuture := workflow.ExecuteChildWorkflow(ctx, SampleChildWorkflow, message)
+
+		selector := workflow.NewSelector(ctx)
+
+		selector.AddFuture(childWorkflowFuture, func(f workflow.Future) {
+			err = f.Get(ctx, &result)
+
+			// we don't want to end the parent workflow when child workflow is cancelled
+			if errors.Is(ctx.Err(), workflow.ErrCanceled) {
+				logger.Info("Child execution cancelled.")
+				return
+			}
+
+			if err != nil {
+				logger.Error("Child execution failure.", "Error", err)
+			}
+
+			// Child workflow completed. Let's end the parent workflow.
+			isProcessingDone = true
+		})
+
+		selector.AddReceive(reBuildSignalChan, func(c workflow.ReceiveChannel, more bool) {
+			logger.Info("Received signal.", "Message", message)
```

Comment:
This log shows the `message` from the _last_ time the loop ran. The message is not received until `GetLatestMessageFromChannel` is executed.